### ISSUE TITLE
Nexus Service Test Failing (v2 Snapshot)

### ISSUE THREAD

**Issue by calebcartwright** [badges/shields#7630]
:clock11: **When did the problem start?**

~9 days ago, Feb 10th

:camera: **Live badge**

https://img.shields.io/nexus/s/com.google.guava/guava?server=https%3A%2F%2Foss.sonatype.org
![](https://img.shields.io/nexus/s/com.google.guava/guava?server=https%3A%2F%2Foss.sonatype.org)

:wrench: **Is the live badge working?**

:link: **CircleCI link**

https://app.circleci.com/pipelines/github/badges/daily-tests/1508/workflows/81f4046f-5d62-45eb-be29-9d93e85e00fd/jobs/2410/tests#failed-test-5

:lady_beetle: **Stack trace**

```
Nexus [live] Nexus 2 - search snapshot version valid snapshot artifact [ GET /s/org.fusesource.apollo/apollo-karaf-feature.json?server=https://repository.jboss.org/nexus ]
[ GET /s/org.fusesource.apollo/apollo-karaf-feature.json?server=https://repository.jboss.org/nexus ]
/home/circleci/project/shields/core/service-test-runner/cli.js

ValidationError: message mismatch: "value" with value "artifact not found" fails to match the required pattern: /^v\d+(\.\d+)*([-+~].*)?$/
    at Object.exports.process (node_modules/joi/lib/errors.js:193:16)
    at Object.internals.entry (node_modules/joi/lib/validator.js:153:26)
    at Object.exports.entry (node_modules/joi/lib/validator.js:27:30)
    at internals.Base.validate (node_modules/joi/lib/base.js:548:26)
    at Object.internals.assert (node_modules/joi/lib/index.js:225:27)
    at Object.attempt (node_modules/joi/lib/index.js:107:26)
    at Function._expectField (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:89:13)
    at IcedFrisbyNock.<anonymous> (file:///home/circleci/project/shields/core/service-test-runner/icedfrisby-shields.js:70:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:954:10)
    at invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:1003:24)
    at /home/circleci/project/shields/node_modules/icedfrisby/lib/icedfrisby.js:1017:7
    at new Promise (<anonymous>)
    at IcedFrisbyNock._runHooks (node_modules/icedfrisby/lib/icedfrisby.js:976:12)
    at IcedFrisbyNock.run (node_modules/icedfrisby/lib/icedfrisby.js:1276:20)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Context.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:1348:9)
```

:bulb: **Possible solution**

Dunno, error message is showing artifact not found so perhaps it was renamed or removed, or we just need to find another target

**Comment by PyvesB** [comment]
These tests are annoying, there are quite a few of them, and we end up needing to fix one or the other on a fairly regular basis. I'm wondering whether we should switch more of them to use fake data and only keep a couple talking to real Nexus instances.

**Comment by calebcartwright** [comment]
I've been wondering about that lately as well. The only potential sticking point in mind would be whether we've still got anything live that's accounting for the API differences between the different version. However, that's not something we necessarily need to maintain in my mind either (I believe SonarQube has a similar set of circumstances with a significantly different API surface between certain major versions)

**Comment by PyvesB** [comment]
Nexus tests were all fixed in #9933 and seem to be holding up, gonna go ahead and close this issue.
